src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -835,7 +843,21 @@ def get_vm_env_datastore(vm)
 
     def add_disk_to_agent_env(vm, director_disk_cid, device_unit_number)
       env = @agent_env.get_current_env(vm.mob, @datacenter.name)
+
       env['disks']['persistent'][director_disk_cid.raw] = device_unit_number.to_s
+
+      for options in vm.extra_config
+        logger.info("adding disk to env, looking for diskUUID on VM: #{options.key}, #{options.value}")
+        if options.key == 'disk.enableUUID'
+          logger.info("adding disk to env, FOUND disk.enableUUID")
+          if options.value == "TRUE"
+            uuid = vm.persistent_disks[0].backing.uuid.downcase
```

Comment:
Are we sure that this is the correct persistent disk? Seems like this will associate the wrong disk when multiple persistent disks are attached to the VM.

Reply:
I need to change this to go through the entire array of persistent disks as (though this is not commercially used) BOSH supports multiple persistent disks.   But we can be sure that they're BOSH persistent disks and not Kubernetes because the `vm.persistent_disks` method only returns devices that match the vApp property that demarcates BOSH persistent disks.

Reply:
`vm.persistent_disks`  does *NOT only* return devices that match vApp. It returns more than that.
```
device.backing.disk_mode == Vim::Vm::Device::VirtualDiskOption::DiskMode::INDEPENDENT_PERSISTENT ||
            persistent_device_keys.any? { |key| device.key == key }
```
Check the OR condition above ^^

Reply:
Good catch.   So I think the better approach is to introduce a new method accessor (`vm.bosh_persistent_disks` )? that filters to ONLY the BOSH persistent disks.

Reply:
Hmm @EleanorRigby is the `add_disk_to_agent_env` method even useful? Seems like it's only called from `attach_disk` in `cloud.rb` and I'm not sure that it represents a meaningful abstraction. If we inline it we'll have access to the underlying `Device` itself from which we can probably get the UUID more reliably.